rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -1474,43 +1591,57 @@ extern "C"
             rmw_guard_conditions_t *guard_conditions,
             rmw_services_t *services,
             rmw_clients_t *clients,
+            rmw_waitset_t * waitset,
             const rmw_time_t *wait_timeout)
     {
-        std::mutex conditionMutex;
-        std::condition_variable conditionVariable;
+        if (!waitset) {
+            return RMW_RET_ERROR;
```

Comment:
When returning an error code it must set the error message before using `RMW_SET_ERROR_MSG`.

Same below.